### ISSUE TITLE
Improve frontalized ffhq

### ISSUE THREAD

**Issue by duongquangvinh** [eladrich/pixel2style2pixel#118]
Dear sir,

It is me again!
I come from #112 and I got the results that seem to be a little bit better (with 140k iterations training for frontalize_ffhq). But always, some results are blur and not too shaped on the boundary of face.

![00403](https://user-images.githubusercontent.com/59604464/114338362-1fb36580-9b7d-11eb-9a2e-af50dfe1ddd0.jpg)
![03254](https://user-images.githubusercontent.com/59604464/114338420-3ce83400-9b7d-11eb-8164-422cabcbaeaf.jpg)

But I had an idea that I do not know whether it is good or not. So I need your advice.
I am going to use one of 2 methods: super resolution or styleGAN encoding to improve the sharpness of inference frontalized FFHQ.

+ In super resolution: I will set source = 'frontalized ffhq', target = 'frontalized ffhq', using pretrained generator stylegan_config_f.
+ In styleGAN encoding: I will set source = 'frontalized ffhq', target = 'original corresponding ffhq', using pretrained generator stylegan_config_f.

Could you give me some advice? Which method do you think better?

**Comment by yuval-alaluf** [comment]
Hi @duongquangvinh,
The results do look good in my opinion. If you want sharper images, I am not sure that the super resolution will help. 
However, if you pass the resulting image through our pretrained encoder (for ffhq_encode) , it will be interesting to see if you get sharper images. Note that what I am referring to here is not training (as you mentioned in your second bullet), but simply passing the image as you would in inference. 
Does this make sense?

**Comment by duongquangvinh** [comment]
Oh thank you for your reply sir!

So you mean I will use your pretrained_styleGAN_encode in inference to get what I want, right?
There are some more points I see from your paper, you said: "In particular, we decrease the LPIPS and L2 loss functions, setting λ1 = 0.01, λ2 = 0.8 over the inner part of the face and λ1 = 0.001, λ2 = 0.08 elsewhere, focusing the model on the inner region while reducing the importance of background preservation". So in the training options for frontalize_ffhq, it has 2 flags --lpips_lambda_crop=0.8, --l2_lambda_crop=0.01. So can we increase a little these configs to enlarge the facial region details (including facial boundary, not only inner part)?

Thank you, sir.

**Comment by yuval-alaluf** [comment]
Yes.  
Regarding the losses, you can try to play with the losses to see if they affect the sharpness of the results.

**Comment by duongquangvinh** [comment]
That is so nice to receive your feedback, sir!
I will post the results after training. Thank you so much, sir!

**Comment by duongquangvinh** [comment]
@yuval-alaluf Dear sir!

This is my result from inference of ffhq_encode and super_resolution. Both use input images as frontalized ffhq images.
The left one is using pretrained super resolution model weight given in repo. The right one uses pretrained ffhq_encode model weight given in repo.

![Screenshot from 2021-04-12 17-55-45](https://user-images.githubusercontent.com/59604464/114384165-9457c580-9bb8-11eb-9f27-d908a8bd7aec.png)
![Screenshot from 2021-04-12 17-56-48](https://user-images.githubusercontent.com/59604464/114384172-96218900-9bb8-11eb-88a6-d834b07a3a03.png)
![Screenshot from 2021-04-12 17-57-22](https://user-images.githubusercontent.com/59604464/114384174-96ba1f80-9bb8-11eb-8a45-f33a21de82a8.png)

Both improves the sharpness of input images. But the left one (super resolution) seem to outweigh the right one. However, it seems to generate non-Asian face as I want. Therefore, I am going to retrain super resolution but I am confused in 2 terms:
+ source = target = 'original FFHQ'
+ source = target = 'frontalized FFHQ'
I do not know whether which one I should choose? Thank you sir!

**Comment by yuval-alaluf** [comment]
As I mentioned in my previous comment, the super resolution model is not really what you want. It is not surprising that the model does not preserve the "Asian appearance" of the input. 
The results of the ffhq_encode model seem excellent (you didn't show the original input, but the sharpness and identity seem good).

**Comment by duongquangvinh** [comment]
Thank you for your reply sir, I also try to use style-mixing feature to generate more image. But the generated images seem to be random (not always Asian). Therefore, I think I have to filter manually.

Thank you sir. We can do so many things with pSp. I hope you can gain more appreciation. I gave you 1 star.

**Comment by yuval-alaluf** [comment]
Thank you for the kind words about pSp. 
Best of luck with your project! And thanks for the star :smile:

**Comment by netuserjun** [comment]
Hello, @duongquangvinh 
I also doing similar project, but my hardware env is not good. 
Can you share the trained model?
